### PR TITLE
Bring NCEPLIBS up-to-date with hpc-stack

### LINKED ISSUE #173: Release version 1.3.0 to match hpc-stack-1.1.0
We are releasing a new version of hpc-stack, and it involves some new versions of some of the NCEPLIBS. So we need a matching release of NCEPLIBS.

See https://github.com/NOAA-EMC/hpc-stack/issues/72

### PR DESCRIPTION
**Pull request by kgerheiser**
Re-name nceppost to upp and update to v10.0.0

Add prod_util

Add g2c

Update wgrib2 to cmake-v6 with bugfixes

Always build ip and ip2. They can co-exist now.

Build wgrib2 executable with USE_IPOLATES and USE_NETCDF4 (wgrib2 builds a simple library, so the library isn't dependent)

Fixes #173

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -29,7 +29,7 @@ set(DEFAULT_LIBCOMPS ${CMAKE_CURRENT_SOURCE_DIR}/COMPONENTS)
 option(OPENMP        "Enable OpenMP threading" OFF)
 option(ENABLE_TESTS  "Enable testing"          OFF)
 option(BUILD_CRTM    "Build CRTM library"      ON)
-option(BUILD_POST    "Build NCEPpost library"  ON)
+option(BUILD_POST    "Build UPP library"  ON)
```
Comment (aerorahul) [inline CMakeLists.txt]: Might as well change this option to BUILD_UPP

**Review by aerorahul — COMMENTED** [review]
Looks good. Just one comment on build option for UPP.

**Review by edwardhartnett — APPROVED** [review]

**Review by climbfuji — APPROVED** [review]
Thanks! I had made those changes locally as well, but didn't create a PR.

File: CMakeLists.txt [hunk 2]
```diff
@@ -29,7 +29,7 @@ set(DEFAULT_LIBCOMPS ${CMAKE_CURRENT_SOURCE_DIR}/COMPONENTS)
 option(OPENMP        "Enable OpenMP threading" OFF)
 option(ENABLE_TESTS  "Enable testing"          OFF)
 option(BUILD_CRTM    "Build CRTM library"      ON)
-option(BUILD_POST    "Build NCEPpost library"  ON)
+option(BUILD_UPP    "Build UPP library"  ON)
```
Comment (aerorahul) [inline CMakeLists.txt]: mind the space.

**Review by aerorahul — CHANGES_REQUESTED** [review]
I believe you moved nceppost.lua to upp.lua.
That is not sufficient.
You will need to define `upp_ROOT` in the modulefile for `find_package(upp)` to work.
You could delete all other variables e.g. nceppost_, NCEPPOST _or POST_ etc.

**Comment by kgerheiser** [comment]
I changed the module to what's in hpc-stack

**Review by aerorahul — APPROVED** [review]
Looks good. Thanks.

**Pull request merged**